src/core/dwn-error.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * A class that represents a DWN error.
+ */
+export class DwnError extends Error {
+  constructor (public code: string, message: string) {
```

Comment:
just an unimportant nit but going with exotic JS stuff like private/public constructors, public/private constructor variables or static functions in classes increases the amount of exotic code we have to transpile with babel with little value gained for it

Reply:
This is more of a syntax-sugar from TypeScript so that you can declare members more concisely:

https://www.typescriptlang.org/docs/handbook/2/classes.html#parameter-properties

But I can sense your view in general on exotic add-ons to JS.

Reply:
@shamilovtim babel outright erases `public`

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/4887440/218674233-cee53397-9043-448b-8c88-aaade3a435f9.png">